src/routes/upload.ts

```diff
@@ -1,20 +1,29 @@
 import { UPLOAD_PATH } from '../config';
 import { Router } from 'express';
 import { createWriteStream, existsSync, rmSync } from 'fs';
+const { execSync } = require('child_process');
 
 const router = Router();
 
 router.post('/', (req, res) => {
   try {
     req.pipe(req.busboy); // Pipe it trough busboy
-
+    const fileSize = req?.query?.fileSize;
+    console.log(`From upload route: Uploaded File size: ${fileSize}`);
     req.busboy.on('file', (fieldname, file, data) => {
       const filename = data && data.filename ? data.filename : 'upload.raucb';
       console.log(`Upload of '${filename}' started`);
       // Create a write stream of the new file
       try {
         try {
           if (existsSync(UPLOAD_PATH + filename)) {
+            const existingFileSize = parseInt(execSync(`ls -l ${UPLOAD_PATH+filename} | awk '{print $5}'`));
+            console.log(`File size is ${existingFileSize} bytes`);
+            if (fileSize && typeof fileSize === 'string' && parseInt(fileSize) === existingFileSize) {
```

Comment:
I think slightly better API would be to pass `size` as number via POST body (req.body.size) or via data fields that you use above (data.size),
this way we will be able to pass the number, and no need to workaround it with strings.